test/test-rotateElement.js

```diff
@@ -0,0 +1,390 @@
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            X----
+            -----
+        `)
+        let pivot = tetris.getCoordinates(`
+            -----
+            -----
+            --X--
+            -----
+            -----
+        `)
+
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -----
+            ---X-
+        `)
+    })
+
+    it("element move the right field 5x5 ", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -----
+            ---X-
+        `)
+        let pivot = tetris.getCoordinates(`
+            -----
+            -----
+            --X--
+            -----
+            -----
+        `)
+
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            -----
+            ----X
+            -----
+            -----
+            -----
+        `)
+    })
+})
+describe("Rotation where starts to the bottom, pivot has different x and y coordinates", function () {
+    it("element move the left field 6x5 ", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -----
+            -----
+            -X---
+        `)
+        let pivot = tetris.getCoordinates(`
+            -----
+            -----
+            -X---
+            -----
+            -----
+            -----
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            -----
+            -----
+            ----X
+            -----
+            -----
+            -----
+        `)
+    })
+
+    it("element move the top field 6x5 ", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            ----X
+            -----
+            -----
+            -----
+        `)
+        let pivot = tetris.getCoordinates(`
+            -----
+            -----
+            -X---
+            -----
+            -----
+            -----
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = [{ x: 1, y: -1 }]
+    })
+
+    it("element move the top field 6x5 ", function () {
+        let elementTetromino = [{ x: 1, y: -1 }]
+        let pivot = tetris.getCoordinates(`
+            -----
+            -----
+            -X---
+            -----
+            -----
+            -----
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = [{ x: 1, y: -1 }]
```

Comment:
this is the same as elementTetromino, incorrect

Reply:
This whole test is just incorrect version of the next test

Reply:
does the error start with line 159 or 185?
I don't understand which test is wrong

Reply:
Remove test on line 340 - it is duplicate of the next test

Reply:
error in line 354